src/lab3/assignment.md

```diff
@@ -1,123 +1,167 @@
+| `city_id`   | number    | A unique ID of the city where the check-in was made. |
+| `city_name` | string    | The name of the city where the check-in was made.    |
+| `refugees`  | number    | The number of refugees entering the vessel.          |
+
+*Disclaimer:* While technically not required, we include the city name to
+facilitate debugging. This goes for some requirements below as well.
+
+The requirements of the Kafka application that you must implement based on this
+description are described below in three tiers ("adequate", "good" and "
+excellent"). These tiers correspond to one of the grading criteria (explained on
+the rubric page). To pass the lab, you must at least achieve the "adequate"
+tier. "Good" and "excellent" tier applications contain **additional**
+requirements (and may supersede previous requirements where applicable) that the
+application needs to adhere to.
+
+### "Adequate" application requirements
+
+1. The application outputs a Kafka stream on the topic `updates` with the total
+   number of refugees in Waterworld, whenever an incoming entry event is pulled
+   from the `events` input stream. The `update` data is serialized as a JSON
+   document with the following schema:
+   
+   | Field       | JSON Type | Description                   |
+   | ----------- | --------- | ----------------------------- |
+   | `refugees`  | number    | The total number of refugees. |
+
+- Example:
+
+On the input stream, the following events are received:
+```json
+{"timestamp":1, "city_id":1, "city_name":"Delft", "refugees":10}
+{"timestamp":3, "city_id":1, "city_name":"Delft", "refugees":20}
+{"timestamp":4, "city_id":2, "city_name":"Rotterdam", "refugees":30}
+```
+On the output stream, the following updates are required:
+```json
+{"refugees":10}
+{"refugees":30}
+{"refugees":60}
+```
 
-Suppose we are beer connoisseurs and we are interested to learn where a lot of
-check-in activity is coming from. Usually this means there is a beer festival
-that we want to visit, and perhaps it is in our city or a city close-by. We do
-want to make sure that there is plenty of beer with styles to our preference. So
-we can filter out or include some of the more [adventurous
-styles](https://en.wikipedia.org/wiki/Gueuze)\*.
+### "Good" application requirements
 
-Your task is to filter check-ins by beer style, and count the number of
-check-ins per city within a time window, and produce streaming updates for the
-real-time world map of the web interface.
+2. Superseding requirement 1, the application outputs the total number of
+   refugees in Waterworld ***for each city***. The `update` data is serialized
+   as a JSON document with the following schema:
 
-### Supplying beer styles
+   | Field       | JSON Type | Description                   | 
+   | ----------- | --------- | ----------------------------- |
+   | `city_id`   | number    | The total number of refugees. |
+   | `city_name` | string    | The name of the city.         |
```

Comment:
The output stream doesn't have to provide the `city_name`. It's dereferenced in the consumer based on the `city_id`. But maybe we keep it there to make it easier to debug the output stream? What do you think?

Reply:
Yeah this was indeed the intention, so I think we should keep it for debugging. (it's mentioned where the input schema is described)